src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuestionStatisticResource.java

```diff
@@ -0,0 +1,128 @@
+
+    private static final String ENTITY_NAME = "questionStatistic";
+
+    private final QuestionStatisticRepository questionStatisticRepository;
+
+    public QuestionStatisticResource(QuestionStatisticRepository questionStatisticRepository) {
+        this.questionStatisticRepository = questionStatisticRepository;
+    }
+
+    /**
+     * POST  /question-statistics : Create a new questionStatistic.
+     *
+     * @param questionStatistic the questionStatistic to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new questionStatistic, or with status 400 (Bad Request) if the questionStatistic has already an ID
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/question-statistics")
+    @Timed
+    public ResponseEntity<QuestionStatistic> createQuestionStatistic(@RequestBody QuestionStatistic questionStatistic) throws URISyntaxException {
+        log.debug("REST request to save QuestionStatistic : {}", questionStatistic);
+        if (questionStatistic.getId() != null) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(ENTITY_NAME, "idexists", "A new questionStatistic cannot already have an ID")).body(null);
+        }
+        QuestionStatistic result = questionStatisticRepository.save(questionStatistic);
+        return ResponseEntity.created(new URI("/api/question-statistics/" + result.getId()))
+            .headers(HeaderUtil.createEntityCreationAlert(ENTITY_NAME, result.getId().toString()))
+            .body(result);
+    }
+
+    /**
+     * PUT  /question-statistics : Updates an existing questionStatistic.
+     *
+     * @param questionStatistic the questionStatistic to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated questionStatistic,
+     * or with status 400 (Bad Request) if the questionStatistic is not valid,
+     * or with status 500 (Internal Server Error) if the questionStatistic couldn't be updated
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PutMapping("/question-statistics")
+    @Timed
+    public ResponseEntity<QuestionStatistic> updateQuestionStatistic(@RequestBody QuestionStatistic questionStatistic) throws URISyntaxException {
+        log.debug("REST request to update QuestionStatistic : {}", questionStatistic);
+        if (questionStatistic.getId() == null) {
+            return createQuestionStatistic(questionStatistic);
+        }
+        QuestionStatistic result = questionStatisticRepository.save(questionStatistic);
+        return ResponseEntity.ok()
+            .headers(HeaderUtil.createEntityUpdateAlert(ENTITY_NAME, questionStatistic.getId().toString()))
+            .body(result);
+    }
+
+    /**
+     * GET  /question-statistics : get all the questionStatistics.
+     *
+     * @param filter the filter of the request
```

Comment:
what kind of filter are you using here? Please add more documentation

Reply:
this was generated automatically

Reply:
ok, I have never seen something like this before in the ArTEMiS code, therefore I was confused ;-)